src/Observable.ts

```diff
@@ -42,7 +39,9 @@ export class Observable<EventSpec> {
     if (typeof arg0 === 'object') {
       // one object with key/value pairs was passed
       for (const eventName in arg0) {
-        this.on(eventName as K, arg0[eventName]);
+        const event = eventName as string as K; // Technically eventName can be also a number or symbol
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        this.on<K, EventSpec[K]>(event, arg0[event]!);
```

Comment:
I dislike this syntax.
`event` is a confusing name in this context, I also prefer inlining it of possible.
Does `K` become a string after applying my first comment? If so it resolves all this.

Reply:
> Does K become a string after applying my first comment? If so it resolves all this.

When I tried yesterday, no. I didn't investigate further though. 

Anyway it would still be:

```ts
this.on<K, EventSpec[K]>(eventName as K, arg0[eventName as K]!);
```

Are you okay with that? For me all that typing makes the code less readable than `this.on<K, EventSpec[K]>(event, arg0[event]!)` but it's up to you.